### DISCUSSION TITLE
Virtual scroll list

### DISCUSSION THREAD

**Discussion by vojty** [emilk/egui#228] [Ideas]
Hi, do you plan to do something like a virtual scroll list for large data sets? Similar to this https://github.com/ocornut/imgui_club#imgui_memory_editor or https://react-window.now.sh/#/examples/list/fixed-size

Basically you need to setup these
- item size
- items count
- visible area

and then provide a callback which decides, what to render (based on current scroll position or current index)

**Comment by emilk** [comment]
Yeah, we definitely want something like this for `egui`, but I haven't yet taken the time to come up with a good design for it. Feel free to open an issue on it!

**Reply by sergchan** [reply]
Hi,
Did you find any scroll list design? any ideas?

**Reply by AlienRecall** [reply]
Hi I'm pretty new to egui so I'm finding all the feature that it has, can I ask if has the virtual scroll been implemented?

**Reply by coderedart** [reply]
aye, you need to use https://docs.rs/egui/latest/egui/containers/struct.ScrollArea.html#method.show_rows or https://docs.rs/egui/latest/egui/containers/struct.ScrollArea.html#method.show_viewport . 

in the demo at https://www.egui.rs/ -> enable scrolling checkbox on the right in the demos list to display the example. at the bottom of that window there's link to the source code in the repo too.

**Reply by AlienRecall** [reply]
> aye, you need to use https://docs.rs/egui/latest/egui/containers/struct.ScrollArea.html#method.show_rows or https://docs.rs/egui/latest/egui/containers/struct.ScrollArea.html#method.show_viewport .
> 
> in the demo at https://www.egui.rs/ -> enable scrolling checkbox on the right in the demos list to display the example. at the bottom of that window there's link to the source code in the repo too.

awesome, thank you
